capture_test.go

```diff
@@ -145,23 +156,108 @@ func TestRerun(t *testing.T) {
 		t.Errorf("unexpected error: %v", err)
 	}
 
-	if want.Output != got.Output {
-		t.Errorf("mismatch in output: wanted %s, got %s", want.Output, got.Output)
-	}
+	AllMatcher(t, want, got)
+}
 
-	if want.ExitCode != got.ExitCode {
-		t.Errorf("mismatch in exitcode: wanted %d, got %d", want.ExitCode, got.ExitCode)
-	}
+func TestCapture_Marshaling(t *testing.T) {
+	cmd := command.New(command.WithEnv([]string{"A=A"}), command.WithDir("/tmp"))
 
-	if !reflect.DeepEqual(want.Args, got.Args) {
-		t.Errorf("mismatch in args: wanted %v, got %v", want.Args, got.Args)
+	want, err := cmd.Run(context.Background(), "ls", "-la")
+	if err != nil {
+		t.Fatalf("expected no error")
 	}
 
-	if want.Name != got.Name {
-		t.Errorf("mismatch in name: wanted %s, got %s", want.Name, got.Name)
+	var marshaled []byte
+
+	marshaled, err = json.Marshal(want)
+	if err != nil {
+		t.Fatalf("Marshal(): expected no error")
 	}
 
-	if want.Dir != got.Dir {
-		t.Errorf("mismatch in dir: wanted %s, got %s", want.Dir, got.Dir)
+	var got command.Capture
+
+	err = json.Unmarshal(marshaled, &got)
+	if err != nil {
+		t.Fatalf("Unmarshal(): expected no error")
 	}
+
+	AllMatcher(t, want, got)
+}
+
+func AllMatcher(t *testing.T, want, got command.Capture) {
+	ExitCodeMatcher(t, want, got)
+	OutputMatcher(t, want, got)
+	ArgsMatcher(t, want, got)
+	NameMatcher(t, want, got)
+	DirMatcher(t, want, got)
+}
+
+func OutputMatcher(t *testing.T, want, got command.Capture) {
```

Comment:
This is something I need to start working on. Tests for me are generally a "Just get it done" and move along, so I've never gone back to evaluate if the tests themselves could be made more efficient / clean